File: codes/perl/operator_ternary.pl
use strict;
use warnings;
use JSON;

print("\n# Ternary Operator in Perl\n");

my $CORRECT_ANSWER = "foo";
print("correct answer: \"$CORRECT_ANSWER\"\n");

my $my_answer;
my $is_my_answer_correct;

$my_answer = 'hello';
print("my answer: \"$my_answer\"\n");
$is_my_answer_correct = $my_answer eq $CORRECT_ANSWER ? "congrats!" : "wrong answer!";
print("result: $is_my_answer_correct\n");
# result: wrong answer!

$my_answer = 'foo';
print("my answer: \"$my_answer\"\n");
$is_my_answer_correct = $my_answer eq $CORRECT_ANSWER ? "congrats!" : "wrong answer!";
print("result: $is_my_answer_correct\n");
# result: congrats!

File: codes/python/operator_ternary.py
print('\n# Ternary Operator in Python')

CORRECT_ANSWER = 'foo'
print(f'correct answer: "{CORRECT_ANSWER}"')

my_answer = 'hello'
print(f'my answer: "{my_answer}"')
is_my_answer_correct = 'congrats!' if my_answer == CORRECT_ANSWER else 'wrong answer!'
print(f'result: {is_my_answer_correct}')
# result: wrong answer!

my_answer = 'foo'
print(f'my answer: "{my_answer}"')
is_my_answer_correct = 'congrats!' if my_answer == CORRECT_ANSWER else 'wrong answer!'
print(f'result: {is_my_answer_correct}')
# result: congrats!
